src/eureka/S5_lightcurve_fitting/models/FleckModel.py

```diff
@@ -0,0 +1,328 @@
+                        star_inc = self.parameters.dict[key][0]
+                    elif 'npts' in split_key[0]:
+                        # it's the number of points to evaluate
+                        npoints = self.parameters.dict[key][0]
+            
+                # Set limb darkening parameters
+                uarray = []
+                for u in self.coeffs:
+                    index = np.where(np.array(self.paramtitles) == u)[0]
+                    if len(index) != 0:
+                        item = self.longparamlist[chan][index[0]]
+                        uarray.append(self.parameters.dict[item][0])
+                pl_params.u = uarray
+                pl_params.limb_dark = self.parameters.dict['limb_dark'][0]
+
+                # Enforce physicality to avoid crashes from batman by returning
+                # something that should be a horrible fit
+                if not ((0 < pl_params.per) and (0 < pl_params.inc < 90) and
+                        (1 < pl_params.a) and (0 <= pl_params.ecc < 1)):
+                    # Returning nans or infs breaks the fits, so this was the
+                    # best I could think of
+                    light_curve = 1e12*np.ma.ones(time.shape)
+                    continue
+
+                # Use batman ld_profile name
+                if self.parameters.limb_dark.value == '4-parameter':
+                    pl_params.limb_dark = 'nonlinear'
+                elif self.parameters.limb_dark.value == 'kipping2013':
+                    # Enforce physicality to avoid crashes from batman by
+                    # returning something that should be a horrible fit
+                    if pl_params.u[0] <= 0:
+                        # Returning nans or infs breaks the fits, so this was
+                        # the best I could think of
+                        lcfinal = 1e12*np.ma.ones(time.shape)
+                        continue
+                    pl_params.limb_dark = 'quadratic'
+                    u1 = 2*np.sqrt(pl_params.u[0])*pl_params.u[1]
+                    u2 = np.sqrt(pl_params.u[0])*(1-2*pl_params.u[1])
+                    pl_params.u = np.array([u1, u2])
+
+                # Make the star object
+                star = fleck.Star(spot_contrast=spot_contrast, 
+                                  u_ld=pl_params.u, 
+                                  rotation_period=star_rotation)
+                # Make the transit model
+                fleck_times = np.linspace(time[0], time[-1], npoints)
+                fleck_transit = star.light_curve(
+                    self.spotlon[chan][:nspots, None]*unit.deg, 
+                    self.spotlat[chan][:nspots, None]*unit.deg, 
+                    self.spotrad[chan][:nspots, None],
+                    star_inc*unit.deg, 
+                    planet=pl_params, times=fleck_times).flatten()
+                m_transit = np.interp(time, fleck_times, fleck_transit)
+                
+                light_curve *= m_transit/m_transit[0]
```

Comment:
The choice to normalize by the first integration rather than the mean or something else seems non-trivial to me. I guess it probably doesn't matter much though since that should presumably be taken care of by the `c0` parameter.

Reply:
yeah, doesn't matter what you normalize by, just needs to be normalized because fleck changes the out of transit baseline based on the spot coverage. c0 takes care of it.